app/controllers/undone_actions_controller.rb

```diff
@@ -0,0 +1,56 @@
+class UndoneActionsController < ApplicationController
+    before_action :authenticate_user!
+    before_action :set_undone_action, only: [:edit, :update, :destroy]
+    before_action :check_role, only: [:edit, :update, :destroy]
+
+    def new
+        @undone = current_user.undone_actions.build
+    end
+    def create
+        @undone = current_user.undone_actions.build(undone_actions_params)
+        if @undone.save
+            flash[:notice] = "UndoneAction was successfully created."
+            redirect_to root_url
+        else
+            flash[:error] = "Failed"
+            render 'new'
+        end
+    end
+    def edit
+        @undone = current_user.undone_actions.find_by(id: params[:id])
+    end
+    def update
+        @undone = current_user.undone_actions.find_by(id: params[:id])
+
+        if @undone.update(undone_actions_params)
+            flash[:notice] = "UndoneAction was successfully updated."
+            redirect_to root_url
+        else
+            flash[:error] = "Failed"
+            render 'edit'
+        end
+    end
+    def destroy
+        @undone = current_user.undone_actions.find_by(id: params[:id])
+        @undone.destroy
+        flash[:notice] = "UndoneAction was successfully destroyed"
+        redirect_to root_url
+     end
+
+
+
+    private
+
+    def set_undone_action
+        @undone = UndoneAction.find_by(id: params[:id])
```

Comment:
find_byだと見つからなくてもエラーにならずにnilが格納されるので、findの方がいいですね。
`@undone`にnilが入ることを想定した実装であればコメントは無視してください。

```suggestion
        @undone = UndoneAction.find(params[:id])
```